### PR TITLE
Signup and login

### PR DESCRIPTION
**Pull request by ShirAmit96**

### CODE REVIEW

File: backend/routes/users.js [hunk 1]
```diff
@@ -1,22 +1,59 @@
...
+  User.find() //list of all users in User model
+    .then((users) => res.json(users))
+    .catch((err) => res.status(400).json("Error: " + err));
 });
 
-router.route('/add').post((req, res) => {
-    const username = req.body.name;
-    const mail = req.body.email;
+function emailRegex(input) {
+  let regex = /[a-zA-Z]+@khealth\.com/i;
```
Comment (ArnonEilat) [inline backend/routes/users.js:11]:
```suggestion
  const regex = /[a-zA-Z]+@khealth\.com/i;
```

File: backend/routes/users.js [hunk 2]
```diff
@@ -1,22 +1,59 @@
...
+router.route("/login").post((req, res) => {
+  const mail = req.body.email;
+  const password = req.body.password;
+  User.find({ mail: mail }, async (err, result) => {
+    if (err) {
+      return res.status(500);
+    }
+    console.log(result);
+    if (result.length === 0) {
+      return res.status(400).json({ line: "Email is wrong. Please try again" });
```
Comment (ArnonEilat) [inline backend/routes/users.js:24]: Better to be consistent with the `return` before `res.status(...`.

I like it like with the return but it should be consistent on all the `res.status(...`.

File: backend/routes/users.js [hunk 3]
```diff
@@ -1,22 +1,59 @@
...
+    newUser
+      .save()
+      .then(() => res.json(newUser._id))
+      .catch((err) => res.status(400).json("Error: " + err));
+  } else if (!emailRegex(mail)) {
+    res.status(400).json({ line: "Invalid Email. Please try again" });
+  } else if (username.length < 2) {
+    res.status(400).json({ line: "User name is too short. Please try again." });
+  } else if (password.length < 8) {
+    console.log("rrr3");
```
Comment (ArnonEilat) [inline backend/routes/users.js:52]:
```suggestion
    console.log("rrr4");
```

File: frontend/src/Pages/Login.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import { Link } from "react-router-dom";
+
+function Login() {
+  const dispatch = useDispatch();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [data, setData] = useState(null);
+  const navigate = useNavigate();
+
+  const login = () => {
```
Comment (ArnonEilat) [inline frontend/src/Pages/Login.js:17]: useEffect for `email` and `password`?

https://reactjs.org/docs/hooks-effect.html

File: frontend/src/Pages/Signup.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      <div className="upperArea2">
+        <Link to="/Login">
+          <img src={Arrow} className="Arrow2" />
+        </Link>
+        <h1 className="headline2">Sign up</h1>
+      </div>
+      <div className="loginArea">
+        <p className="secondHead">Sign up for office registration</p>
+        <input
+          onChange={(n) => setName(n.target.value)}
```
Comment (ArnonEilat) [inline frontend/src/Pages/Signup.js:50]:
```suggestion
          onChange={(e) => setName(e.target.value)}
```

like the others

File: frontend/src/redux/userSlice.js [hunk 6]
```diff
@@ -4,19 +4,19 @@ export const userSlice= createSlice({
   //the name of the slice:
     name:"user",
     initialState: {
-        user:{name:" ",mail:" ",id:" "}
+        user:{name:" ",mail:" ",password:" ",id:" "}
```
Comment (ArnonEilat) [inline frontend/src/redux/userSlice.js:7]:
```suggestion
        user: { name:"", mail: "", password: "", id: "" }
```

**Review by ArnonEilat — APPROVED** [review]
👑

File: backend/routes/users.js (hunk 3 above)
Comment (ShirAmit96) [reply backend/routes/users.js:52]: i deleted this line. it was for debugging.

File: backend/routes/users.js (hunk 2 above)
Comment (ShirAmit96) [reply backend/routes/users.js:24]: added "return" to all relevant lines.

**Pull request merged**